src/Runtime/Distributions/Automata/Automaton.cs

```diff
@@ -144,6 +144,20 @@ protected Automaton()
         /// a need to override the actual automaton value. 
         /// </remarks>
         public double? LogValueOverride { get; set; }
+        
+        /// <summary>
+        /// Whether there are log value overrides at the element level.
+        /// </summary>
+        public bool HasElementLogValueOverrides
```

Comment:
Since this is DiscreteChar specific, does it make sense to move it to StringAutomaton?

Reply:
Good point. I have moved it to StringAutomaton.